Compatibility with GHC 9.2

Could you upload a compatible release?

With `--allow-newer`, I currently get an error:

```
src/Snap/Internal/Parsing.hs:439:64: error:
    • Couldn't match expected type ‘ghc-prim-0.8.0:GHC.Prim.Word#’
                  with actual type ‘ghc-prim-0.8.0:GHC.Prim.Word8#’
    • In the first argument of ‘uncheckedShiftRL#’, namely ‘a#’
      In the first argument of ‘word2Int#’, namely
        ‘(uncheckedShiftRL# a# b#)’
      In the first argument of ‘I#’, namely
        ‘(word2Int# (uncheckedShiftRL# a# b#))’
    |
439 |     shiftr (W8# a#) (I# b#) = I# (word2Int# (uncheckedShiftRL# a# b#))
    |                                                                ^^
cabal: Failed to build snap-core-1.0.4.2
```

--- a/.gitignore
+++ b/.gitignore
@@ -22,3 +22,5 @@ cabal.sandbox.config
 cabal.project.local
 dist-newstyle
 .ghc.environment.*
+.stack-work/
+stack.yaml.lock

--- a/snap-core.cabal
+++ b/snap-core.cabal
@@ -148,7 +148,7 @@ Library
     readable                  >= 0.1     && < 0.4,
     regex-posix               >= 0.95    && < 1,
     text                      >= 0.11    && < 1.3,
-    time                      >= 1.0     && < 1.11,
+    time                      >= 1.0     && < 1.14,
     transformers              >= 0.3     && < 0.6,
     transformers-base         >= 0.4     && < 0.5,
     unix-compat               >= 0.3     && < 0.6,

--- a/snap-core.cabal
+++ b/snap-core.cabal
@@ -131,7 +131,7 @@ Library
 
   build-depends:
     HUnit                     >= 1.2     && < 2,
-    attoparsec                >= 0.12    && < 0.14,
+    attoparsec                >= 0.12    && < 0.15,
     base                      >= 4       && < 5,
     bytestring                >= 0.9     && < 0.12,
     bytestring-builder        >= 0.10.4  && < 0.11,

--- a/src/Snap/Internal/Parsing.hs
+++ b/src/Snap/Internal/Parsing.hs
@@ -27,7 +27,15 @@ import qualified Data.Map                         as Map (empty, insertWith, toL
 import           Data.Maybe                       (Maybe (..), maybe)
 import           Data.Monoid                      (Monoid (mconcat, mempty), (<>))
 import           Data.Word                        (Word8)
-import           GHC.Exts                         (Int (I#), uncheckedShiftRL#, word2Int#)
+import           GHC.Exts                         ( Int (I#)
+                                                  , word2Int#
+#if MIN_VERSION_base(4,16,0)
+                                                  , uncheckedShiftRLWord8#
+                                                  , word8ToWord#
+#else
+                                                  , uncheckedShiftRL#
+#endif
+                                                  )
 import           GHC.Word                         (Word8 (..))
 import           Prelude                          (Bool (..), Either (..), Enum (fromEnum, toEnum), Eq (..), Num (..), Ord (..), String, and, any, concatMap, elem, error, filter, flip, foldr, fst, id, map, not, otherwise, show, snd, ($), ($!), (&&), (++), (.), (||))
 import           Snap.Internal.Http.Types         (Cookie (Cookie))
@@ -437,6 +445,9 @@ hexd c0 = char8 '%' <> word8 hi <> word8 low
     !hi       = toDigit $ (c .&. 0xf0) `shiftr` 4
 
     shiftr (W8# a#) (I# b#) = I# (word2Int# (uncheckedShiftRL# a# b#))
+#if MIN_VERSION_base(4,16,0)
+    uncheckedShiftRL# a# b# = word8ToWord# (uncheckedShiftRLWord8# a# b#)
+#endif
 
 
 ------------------------------------------------------------------------------

--- a/stack.yaml
+++ b/stack.yaml
@@ -1,0 +1,9 @@
+resolver: lts-18.18
+
+#### Uncomment below to test GHC 9.2 on nightly
+#
+# resolver: nightly-2021-12-15
+# compiler: ghc-9.2.1
+#
+# extra-deps:
+#   - attoparsec-0.14.3

--- a/test/Snap/Util/CORS/Tests.hs
+++ b/test/Snap/Util/CORS/Tests.hs
@@ -8,6 +8,9 @@ module Snap.Util.CORS.Tests (tests) where
 import           Data.ByteString.Char8          (ByteString)
 import           Data.CaseInsensitive           (CI (..))
 import qualified Data.HashSet                   as HashSet
+import qualified Data.Set                       as Set
+import qualified Data.Text                      as Text
+import qualified Data.Text.Encoding             as Text
 import           Snap.Core                      (Method (..), getHeader, Response(..))
 import           Snap.Test                      (RequestBuilder, runHandler, setHeader, setRequestType, RequestType(..), setRequestPath)
 import           Snap.Util.CORS                 (applyCORS,CORSOptions(..),defaultOptions,HashableMethod(..))
@@ -97,7 +100,12 @@ checkExposeHeaders :: Maybe ByteString -> Response -> Assertion
 checkExposeHeaders = checkHeader "Access-Control-Expose-Headers"
 
 checkAllowHeaders :: Maybe ByteString -> Response -> Assertion
-checkAllowHeaders = checkHeader "Access-Control-Allow-Headers"
+checkAllowHeaders v r =
+  assertEqual "Header Access-Control-Allow-Headers"
+    (getSet <$> v)
+    (getSet <$> getHeader "Access-Control-Allow-Headers" r)
+  where
+    getSet = Set.fromList . Text.splitOn ", " . Text.decodeUtf8
 
 checkAllowMethods :: Maybe ByteString -> Response -> Assertion
 checkAllowMethods = checkHeader "Access-Control-Allow-Methods"

